src/Router.ts

```diff
@@ -172,12 +172,89 @@ export class Router<C extends Context> extends Evented implements RouterInterfac
 		}
 	}
 
-	dispatch(context: Context, path: string): Task<DispatchResult> {
-		// Reset, any further calls can't have come from start(). This is necessary since the navstart listeners
-		// may call dispatch() themselves.
-		let dispatchFromStart = this._dispatchFromStart;
+	private _dispatch(context: Context, path: string, canceled: boolean = false, emit: boolean = true) {
+		if (canceled) {
+			return { success: false };
```

Comment:
No canceled test?

Reply:
Is it because the `.dispatch()` catches cancels before this is ever called?

Reply:
You be right, I had noticed this was now completely uncovered :)